### PR TITLE
[Fix] Missing device initializations

### PR DESCRIPTION
**Pull request by alopezgit**
### Description
Some cuda tests in `make test` were failing. To solve it, the code:
1) Fixes missing `device` arguments in some core functions and tests. 
2) Skips some tests that fail due to known issues (#649, #666).

There is also a new jit test added in `test_linalg.py`.

`make test` should not return any failing test now.

### Status
Ready

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [X] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/enhance/adjust.py [hunk 1]
```diff
@@ -445,7 +445,7 @@ def build_lut(histo, step):
         # and then normalization by step.
         lut = (torch.cumsum(histo, 0) + (step // 2)) // step
         # Shift lut, prepending with 0.
-        lut = torch.cat([torch.zeros(1), lut[:-1]])
+        lut = torch.cat([torch.zeros(1, device=lut.device), lut[:-1]])
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:448]: this might need `dtype` too

File: kornia/geometry/transform/imgwarp.py [hunk 2]
```diff
@@ -349,9 +349,11 @@ def get_rotation_matrix2d(
                          .format(center.shape, angle.shape, scale.shape))
     # convert angle and apply scale
     rotation_matrix: torch.Tensor = angle_to_rotation_matrix(angle)
-    scaling_matrix: torch.Tensor = torch.zeros((2, 2)).fill_diagonal_(1).repeat(rotation_matrix.size(0), 1, 1)
+    scaling_matrix: torch.Tensor = torch.zeros(
+        (2, 2), device=rotation_matrix.device, dtype=rotation_matrix.dtype).fill_diagonal_(1).repeat(
+        rotation_matrix.size(0), 1, 1)
```
Comment (edgarriba) [inline kornia/geometry/transform/imgwarp.py:354]: you can use `K.eye_like(2, rotation_matrix)`

File: test/augmentation/test_augmentation_mix.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def test_random_mixup_lam0(self, device, dtype):
 
         assert_allclose(out_image, expected, rtol=1e-4, atol=1e-4)
         assert (out_label[:, 0] == label).all()
-        assert (out_label[:, 1] == torch.tensor([0, 1])).all()
+        assert (out_label[:, 1] == torch.tensor([0, 1], device=device)).all()
```
Comment (edgarriba) [inline test/augmentation/test_augmentation_mix.py:82]: missing `dtype` ?

File: test/geometry/test_linalg.py [hunk 4]
```diff
@@ -79,17 +79,17 @@ def test_gradcheck(self, device):
...
-        points = torch.ones(1, 2, 2).to(device)
-        transform = torch.eye(3)[None].to(device)
+        points = torch.ones(1, 2, 2, device=device)
+        transform = kornia.eye_like(3, points)
+        op = kornia.transform_points
+        op_script = torch.jit.script(op)
         actual = op_script(transform, points)
-        expected = kornia.transform_points(transform, points)
-
+        expected = op(transform, points)
```
Comment (edgarriba) [inline test/geometry/test_linalg.py:92]: great! in next PR, after fixing `transformation_points` remove below test

**Review by edgarriba — APPROVED** [review]
LGTM. Just few suggestions that can be implemented in upcoming PRs
